core/generics/thresholdmap/thresholdmap.go

```diff
@@ -169,7 +169,7 @@ func (t *ThresholdMap[K, V]) wrapNode(node *thresholdmap.Element) (element *Elem
 }
 
 // Encode returns a serialized byte slice of the object.
-func (t ThresholdMap[K, V]) Encode() ([]byte, error) {
+func (t *ThresholdMap[K, V]) Encode() ([]byte, error) {
```

Comment:
The problem is, that if you don't use the pointer receiver here, the mutex is copied, and doesn't work as intended.

Reply:
That's true and a good point. However, we use it embedded into other types that usually do locking as well. Also when removing it doesn't fulfill the serix interface anymore (when not a pointer). Let's keep it like it is for now because it would otherwise break our current codebase. Once we are done with the pruning refactor we should change this

Reply:
I will create an issue and assign you to it.